### ISSUE TITLE
unnecessary_fold is not commutative

### ISSUE THREAD

**Issue by Abraxas-Knister** [rust-lang/rust-clippy#16581]
### Summary

sums and products are commutative, but sum() and product() are only suggested by clippy under some circumstances.

### Lint Name

unnecessary_fold

### Reproducer

I tried this code:

```rust
let six = (2..=3).fold(1,|a,b|a*b); // 2*3
let six = (1..=3).fold(0,|a,b|a+b); // 1+2+3
let six = (2..=3).fold(1,|b,a|a*b); // 3*2
let six = (1..=3).fold(0,|b,a|a+b); // 3+2+1

```
I expected to see this happen: all lines should get annotated for unnecessary_fold.

Instead, this happened: only the first two are.

### Version

```text
rustc 1.93.1 (01f6ddf75 2026-02-11)
binary: rustc
commit-hash: 01f6ddf7588f42ae2d7eb0a2f21d44e8e96674cf
commit-date: 2026-02-11
host: x86_64-unknown-linux-gnu
release: 1.93.1
LLVM version: 21.1.8
```
